internal/controllers/operator_controller.go

```diff
@@ -136,6 +140,21 @@ func (r *OperatorReconciler) reconcile(ctx context.Context, op *operatorsv1alpha
 		return ctrl.Result{}, err
 	}
 
+	// TODO: Checking for unsat is awkward using the current version of deppy.
+	//    This awkwardness has been fixed in an unreleased version of deppy.
+	//    When there is a new minor release of deppy, we can revisit this and
+	//    simplify this to a normal error check.
+	//    See https://github.com/operator-framework/deppy/issues/139.
```

Comment:
On the working group call today, I mentioned that I would try to go ahead and bump deppy to the commit containing this fix.

Unfortunately that commit brings in some breaking changes with deppy's removal of the EntitySource API, so we'll have to wait until operator-controller accounts for that change. (see https://github.com/operator-framework/operator-controller/issues/347)